node/src/blockchain/blockchain_bridge.rs

```diff
@@ -1434,34 +1707,53 @@ mod tests {
     }
 
     #[test]
-    fn processing_of_received_payments_continues_even_if_no_payments_are_detected() {
+    fn handle_retrieve_transactions_receives_invalid_topics() {
         init_test_logging();
-        let lower_interface =
-            LowBlockchainIntMock::default().get_block_number_result(Ok(0u64.into()));
-        let blockchain_interface_mock = BlockchainInterfaceMock::default()
-            .retrieve_transactions_result(Ok(RetrievedBlockchainTransactions {
-                new_start_block: 7,
-                transactions: vec![],
-            }))
-            .lower_interface_results(Box::new(lower_interface));
-        let persistent_config = PersistentConfigurationMock::new()
-            .max_block_count_result(Ok(Some(10000u64)))
-            .start_block_result(Ok(6));
+        let test_name = "handle_retrieve_transactions_receives_invalid_topics";
+        let system = System::new(test_name);
+        let logger = Logger::new(test_name);
+        let port = find_free_port();
+        let expected_response_logs = vec![LogObject {
+            removed: false,
+            log_index: Some("0x20".to_string()),
+            transaction_index: Some("0x30".to_string()),
+            transaction_hash: Some(
+                "0x2222222222222222222222222222222222222222222222222222222222222222".to_string(),
+            ),
+            block_hash: Some(
+                "0x1111111111111111111111111111111111111111111111111111111111111111".to_string(),
+            ),
+            block_number: Some("0x7D0".to_string()), // 2000 decimal
+            address: "0x3333333333333333333333333333333333333334".to_string(),
+            data: "0x000000000000000000000000000000000000000000000000000000003b5dc100".to_string(),
+            topics: vec![
+                "0xddf252ad1be2c89b69c2b0680000000000006561726e696e675f77616c6c6574".to_string(),
+            ],
+        }];
+        let _blockchain_client_server = MBCSBuilder::new(port)
+            .response("0x3B9ACA00".to_string(), 0)
+            .response(expected_response_logs, 1)
+            .start();
         let (accountant, _, accountant_recording_arc) = make_recorder();
-        let system = System::new(
-            "processing_of_received_payments_continues_even_if_no_payments_are_detected",
-        );
+        let accountant_addr =
+            accountant.system_stop_conditions(match_every_type_id!(ReceivedPayments));
+        let earning_wallet = make_wallet("earning_wallet");
+        let mut blockchain_interface = make_blockchain_interface_web3(Some(port));
+        blockchain_interface.logger = logger;
+        let persistent_config = PersistentConfigurationMock::new()
+            .start_block_result(Ok(6))
+            .max_block_count_result(Err(PersistentConfigError::NotPresent));
```

Comment:
This is wrong. In this test scenario there's no place for an error other than from the response to the EthLogsScan. If you still are convinced that this is taking place rightly (as I have a dim picture of what it may do) at least don't use such a weird error like PersistentConfigError::NotPresent. I think its meaning is: "this paramater you are querying doesn't have a row in the config table" which is a serious trouble. You want something else or a None value.

Now I've regained the confidence in my assumption that you're putting an error to a test which already has its crucial error and you shouldn't do this. The test should proabably test only one error case a time.

Reply:
**URGENCY STATUS: AT LEAST THE SECOND PARAGRAPH IS IMPORTANT. CONFUSION DRAGGED IN TEST**